### PR TITLE
MediaQuery.of shouldn't give a default return value

### LINKED ISSUE #7547: MediaQuery.of should not fallback
MediaQuery.of() should throw a FlutterError explaining that you should use a WidgetsApp if it doesn't find a MediaQuery in the tree.

### PR DESCRIPTION
**Pull request by Hixie**
Fixes https://github.com/flutter/flutter/issues/7547

### CODE REVIEW

**Comment by Hixie** [comment]
This will be finished Thursday. Turns out it affects a lot of tests (mostly because I made Scaffold throw if there's no MediaQuery in scope).

**Comment by Hixie** [comment]
cc @abarth

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 1]
```diff
@@ -146,9 +166,27 @@ class MediaQuery extends InheritedWidget {
...
-    return query?.data ?? new MediaQueryData.fromWindow(ui.window);
+    if (query != null)
+      return query.data;
+    if (nullOk)
+      return null;
+    throw new FlutterError(
+      'MediaQuery.of() called with a context that does not contain a MediaQuery.\n'
+      'No MediaQuery ancestor could be found starting from the context that was passed '
+      'to MediaQuery.of(). This can happen because you do not have a [WidgetsApp] or '
+      '[MaterialApp] widget (those widgets introduce a [MediaQuery]), or it can happen '
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/media_query.dart]: Too many [ ] ]  [ ] [ ][] [] ]

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/widgets/media_query.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/media_query.dart]: fixed

**Comment by Hixie** [comment]
will land on green

**Comment by Hixie** [comment]
Landing on red because the only failure is a flake and I've submitted a patch to mark it as flaky (https://github.com/flutter/flutter/pull/9656).

**Pull request merged**